addons/sale/models/sale_order_line.py

```diff
@@ -440,15 +440,15 @@ def _compute_price_unit(self):
             if not line.product_uom or not line.product_id:
                 line.price_unit = 0.0
             else:
-                price = line.with_company(line.company_id)._get_display_price()
-                line.price_unit = line.product_id._get_tax_included_unit_price(
-                    line.company_id,
-                    line.order_id.currency_id,
-                    line.order_id.date_order,
-                    'sale',
+                line = line.with_company(line.company_id)
+                price = line._get_display_price()
+                line.price_unit = line.product_id._get_tax_included_unit_price_from_price(
+                    price,
+                    line.currency_id or line.order_id.currency_id,
+                    product_taxes=line.product_id.taxes_id.filtered(
+                        lambda tax: tax.company_id == line.env.company
```

Comment:
```suggestion
                        lambda tax: tax.company_id == line.company_id
```
Note: the current code is working thanks to `line.with_company(line.company_id)`

Reply:
I know, but that way, at least the customer has the taxes from the env company if you stupidly (and temporarily since you cannot save without it) emptied the order company_id field.